packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt

```diff
@@ -80,4 +92,27 @@ class AppContext(
    */
   val taskManager: TaskManagerInterface?
     get() = legacyModule()
+
+  /**
+   * Provides access to the activity provider from the legacy module registry
+   */
+  val activityProvider: ActivityProvider?
+    get() = legacyModule()
+
+  fun onDestroy() {
+    reactContext.get()?.removeLifecycleEventListener(this)
+    registry.post(EventName.MODULE_DESTROY)
+  }
+
+  override fun onHostResume() {
+    registry.post(EventName.APP_ENTERS_FOREGROUND)
```

Comment:
Hmm, maybe it will be better to make them platform-scepific. I will rename the `APP_ENTERS_FOREGROUND` and `APP_ENTERS_BACKGROUND` to `ACTIVITY...`. The current naming can be confusing for the developers because those events are triggered at different times.